pkg/l2serviceattachmentparser/l2serviceattachmentparser.go

```diff
@@ -45,59 +44,61 @@ func NewL2SrvAttParser(srvAttObj *enov1alpha1.L2ServiceAttachment, l2srvObjs []*
 }
 
 // ParseL2ServiceAttachment - parses a L2ServiceAttachment Resource
-func (sattp *L2SrvAttParser) ParseL2ServiceAttachment(d *render.RenderData) (string, error) {
+func (sattp *L2SrvAttParser) ParseL2ServiceAttachment(data map[string]interface{}) (string, string, error) {
 	// Initiate Parsers
 	cpParser := connectionpointparser.NewCpParser(sattp.cpResource, sattp.log)
 
 	// Parse ConnectionPoint object
-	cpParser.ParseConnectionPoint(d)
+	cpParser.ParseConnectionPoint(data)
 
 	cniToUse, err := sattp.pickCni()
 	if err != nil {
 		sattp.log.Error(err, "Error occurred while picking cni")
-		return "", err
+		return "", "", err
 	}
 
 	cniObj := sattp.cniMapping[cniToUse]
 	cniConfigObj := sattp.instantiateCniConfig()
-	manifestFolder, err := cniObj.HandleCni(cniConfigObj, d)
+	manifestFile, err := cniObj.HandleCni(cniConfigObj, data)
 	if err != nil {
 		sattp.log.Error(err, "Error occured while handling cni")
-		return "", err
+		return "", "", err
 	}
 
+	var ipamManifestFile string
 	if len(sattp.subnetResources) > 0 {
 		var ipType, ipamType string
 		for i, subnet := range sattp.subnetResources {
 			if i > 0 {
 				if ipType == subnet.Spec.Type {
 					err := errors.New("subnets in one L2Service should not be of same ip type")
 					sattp.log.Error(err, "")
-					return "", err
+					return "", "", err
 				}
 				if ipamType != subnet.Spec.Ipam {
 					err := errors.New("subnets in one L2Service should have same ipam type")
 					sattp.log.Error(err, "")
-					return "", err
+					return "", "", err
 				}
 			}
 			ipType = subnet.Spec.Type
 			ipamType = subnet.Spec.Ipam
 			sp := parser.NewSubnetParser(subnet, sattp.log)
 			if err := sp.ValidateSubnet(); err != nil {
-				return "", err
+				return "", "", err
 			}
 			if err := sp.ValidateRoute(sattp.routeResources[subnet.GetName()]); err != nil {
-				return "", err
+				return "", "", err
 			}
 		}
```

Comment:
Can we put all this validation code to some dedicated function and not just having so much code inside a single function ?

Reply:
separated out to have cni and ipam specific funtion.